packages/module/src/DataViewTableTree/DataViewTableTree.tsx

```diff
@@ -0,0 +1,158 @@
+      - currentRowIndex - position of the row relative to the entire table
+      - isHidden - defaults to false, true if this row's parent is expanded
+  */
+  const renderRows = (
+    [ node, ...remainingNodes ]: DataViewTrTree[],
+    level = 1,
+    posinset = 1,
+    rowIndex = 0,
+    isHidden = false
+  ): React.ReactNode[] => {
+    if (!node) {
+      return [];
+    }
+    const isExpanded = expandedNodeIds.includes(node.id);
+    const isDetailsExpanded = expandedDetailsNodeNames.includes(node.id);
+    const isChecked = isNodeChecked(node);
+    let icon = leafIcon;
+    if (node.children) {
+      icon = isExpanded ? expandedIcon : collapsedIcon;
+    }
+
+    const treeRow: TdProps['treeRow'] = {
+      onCollapse: () =>
+        setExpandedNodeIds((prevExpanded) => {
+          const otherExpandedNodeIds = prevExpanded.filter((id) => id !== node.id);
+          return isExpanded ? otherExpandedNodeIds : [ ...otherExpandedNodeIds, node.id ];
+        }),
+      onToggleRowDetails: () =>
+        setExpandedDetailsNodeIds((prevDetailsExpanded) => {
+          const otherDetailsExpandedNodeIds = prevDetailsExpanded.filter((id) => id !== node.id);
+          return isDetailsExpanded ? otherDetailsExpandedNodeIds : [ ...otherDetailsExpandedNodeIds, node.id ];
+        }),
+      onCheckChange: (isSelectDisabled?.(node) || !onSelect) ? undefined : (_event, isChecking) => onSelect?.(isChecking, getDescendants(node)),
+      rowIndex,
+      props: {
+        isExpanded,
+        isDetailsExpanded,
+        isHidden,
+        'aria-level': level,
+        'aria-posinset': posinset,
+        'aria-setsize': node.children?.length ?? 0,
+        isChecked,
+        ouiaId: `${ouiaId}-tree-toggle-${node.id}`,
+        checkboxId: `checkbox_id_${node.id?.toLowerCase().replace(/\s+/g, '_')}`,
+        icon,
+      }
+    };
+
+    const childRows =
+      node.children && node.children.length
+        ? renderRows(node.children, level + 1, 1, rowIndex + 1, !isExpanded || isHidden)
+        : [];
+
+    return [
+      <TreeRowWrapper key={node.id} row={{ props: treeRow.props }}>
+        {node.row.map((cell, colIndex) => {
+          const cellIsObject = isDataViewTdObject(cell);
+          return (
+            <Td
+              key={colIndex}
+              treeRow={colIndex === 0 ? treeRow : undefined}
+              {...(cellIsObject && (cell?.props ?? {}))}
+              data-ouia-component-id={`${ouiaId}-td-${rowIndex}-${colIndex}`}
+            >
+              {cellIsObject ? cell.cell : cell}
+            </Td>
+          )
+        })}
+      </TreeRowWrapper>,
+      ...childRows,
+      ...renderRows(remainingNodes, level, posinset + 1, rowIndex + 1 + childRows.length, isHidden)
+    ];
+  };
+
+  return (
+    <Table isTreeTable aria-label="Data table" ouiaId={ouiaId} {...props}>
+      <DataViewTableHeader isTreeTable columns={columns} ouiaId={ouiaId} />
+      <Tbody>
+        {renderRows(rows)}
```

Comment:
I'd store the node(s) into a variable and just use that instead of calling the function.